### PR TITLE
Treat the budget as a per-second budget

### PR DESCRIPTION
**Pull request by Swatinem**
This will now try (and fail in some edge-case) to properly round the spent budget to a per-second basis.

### CODE REVIEW

File: src/stats.rs [hunk 1]
```diff
@@ -160,6 +179,43 @@ mod tests {
...
+
+        for _ in 0..(5 * 10) {
+            let now = timer.now();
+            let truncated_now = stats.config.truncated_now(now);
+            let spent_budget = stats.spent_budget(now, truncated_now);
+            // FIXME:
+            // Ideally this should round to 100 all the time,
+            // but there still seems to be an edgecase in here somewhere:
+            // This rounds to 100 for 9/10 cases, but ~125 in 1/10.
+            dbg!(spent_budget);
```
Comment (Swatinem) [inline src/stats.rs:215]: @loewenheim maybe you have an idea why this would be happening? You are usually better spotting these things than me.

Maybe instead of truncating down, we should be truncating upwards? Or use round instead of truncate?

Comment (loewenheim) [reply src/stats.rs:215]: I  think the edge case you're hitting here is that if `adjustment` (`now - truncated_now`) is `0`, you are exactly on the edge of a bucket, and then subtracting the `bucket_size` in line 119 is incorrect.

File: src/stats.rs [hunk 2]
```diff
@@ -102,6 +101,25 @@ impl ProjectStats {
...
+            .iter()
+            .filter_map(|b| (b.0 >= earliest_time).then_some(b.1))
+            .sum();
+
+        // The configured budget is meant as a per-second budget.
+        // To calculate that, we want to divide by the real passed time,
+        // to avoid any artifacts resulting from the bucketing as much as possible.
+        let adjustment = now - truncated_now;
+        let adjusted_time_window =
+            self.config.budgeting_window - self.config.bucket_size + adjustment;
```
Comment (loewenheim) [inline src/stats.rs:119]: This should make it work:
```suggestion
        let adjusted_time_window = if adjustment != Duration::ZERO {
            self.config.budgeting_window - self.config.bucket_size + adjustment
        } else {
            self.config.budgeting_window
        };
```

**Review by loewenheim — CHANGES_REQUESTED** [review]

**Review by loewenheim — APPROVED** [review]
